setup.py

```diff
@@ -12,19 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Setup for pip package."""
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
+import os
 
 from setuptools import find_packages
 from setuptools import setup
 from setuptools.dist import Distribution
 
-__version__ = '0.0.1'
+version = {}
+base_dir = os.path.dirname(os.path.abspath(__file__))
+with open(os.path.join(base_dir, "tensorflow_addons", "version.py")) as fp:
+    exec(fp.read(), version)
```

Comment:
Could we use import like https://github.com/tensorflow/probability/blob/80cc5cb33dfe04cb998bfe27ad3680a7a116d8b1/setup.py#L29 ? It seems simpler and cleaner.

Reply:
I picked number 3 from this:
https://packaging.python.org/guides/single-sourcing-package-version/

I'm not sure I agree that modifying system path is cleaner than this. It'll usually only be done in a build container, but still don't love the idea.

Reply:
I'm just worried that `exec` is not a safe function. And only in the process of setup.py, system path is changed. It seems not so bad for me.

Run it twice, we get always the same result:
```
/usr/bin/python -c "import sys; print(sys.path); sys.path.append('hello'); print(sys.path)"
```

Anyway, I understand your concern. Both solutions are fine to me.